CNV-19775: configuring the default CPU model (#60760)
openshift/openshift-docs: modules/virt-configuring-default-cpu-model.adoc

// Module included in the following assemblies:
//
// * virt/virtual_machines/vm_networking/virt-configuring-default-cpu-model.adoc

:_content-type: PROCEDURE
[id="virt-configuring-default-cpu-model_{context}"]
= Configuring the default CPU model

> Configure the `defaultCPUModel` by updating the `HyperConverged` custom resource (CR). You can change the `defaultCPUModel` while OpenShift Virtualization is running.

Reviewer: I think this should use the attribute `:VirtProductName:` right?
```suggestion
Configure the `defaultCPUModel` by updating the `HyperConverged` custom resource (CR). You can change the `defaultCPUModel` while {VirtProductName} is running.
```

Author: Indeed!

> Note: The `defaultCPUModel` is case sensitive.

Reviewer: this should be formatted like other asciidoc admonitions across the docs using block syntax:
```suggestion
[NOTE]
====
The `defaultCPUModel` is case sensitive.
====
```

.Prerequisites

* Install the OpenShift CLI (oc).

.Procedure

. Open the `HyperConverged` CR by running the following command:
+
[source,terminal]
----

> $ oc edit hco -n openshift-cnv kubevirt-hyperconverged

Reviewer 2: Can we be sure HCO is always installed in this namespace?
Maybe it's worthwhile to mention that the namespace should the namespace on which Openshift Virtualization (CNV) is installed.

But I may be over-thinking this.

----

. Add the `defaultCPUModel` field to the CR and set the value to the name of a CPU model that exists in the cluster:

+
[source,yaml]
----
apiVersion: hco.kubevirt.io/v1beta1
kind: HyperConverged
metadata:
 name: kubevirt-hyperconverged
 namespace: openshift-cnv
spec:
  defaultCPUModel: "EPYC"
----

> . Apply the YAML file to your cluster.

Reviewer: normally there is a command at this step to apply config files to the cluster, something like:
```suggestion
. Apply the YAML file to your cluster:
[source,terminal]
----
$ oc apply -f <config-name>.yaml
----
```
i'm not totally sure how it would look for your use case, but an SME would know for sure 🙂

Author: @iholder101 Please review the code sample here and let me know if it requires any update or is good to ad to the doc as-is.

Reviewer 2: I don't think that's necessary as you already provide a command [here](https://github.com/openshift/openshift-docs/pull/60760/files#diff-9dd043aed0a1fd14f9c9175ba0d222618c45195515b39fb68a9c0827a3305912R26):

```
oc edit hco -n openshift-cnv kubevirt-hyperconverged
```

Author: Thanks @iholder101 !
